src/lib/helpers.ts

```diff
@@ -236,3 +236,42 @@ export function retryable<T>(
 
   return promise
 }
+
+function dec2hex(dec: number) {
+  return ('0' + dec.toString(16)).substr(-2)
+}
+
+// Functions below taken from: https://stackoverflow.com/questions/63309409/creating-a-code-verifier-and-challenge-for-pkce-auth-on-spotify-api-in-reactjs
+export async function generatePKCEVerifier() {
+  const verifierLength = 56
+  const array = new Uint32Array(verifierLength)
+  if (!isBrowser()) {
+    for (let i = 0; i < verifierLength; i++) {
+      array[i] = Math.floor(Math.random() * 256)
+    }
+  }
```

Comment:
@J0 
Trying to implement PKCE in Flutter library as well, and had a question here. 

What is the reason why we need to limit this to only run in browser?

Reply:
Are these codes expected to ever run in a server environment?

Reply:
Makes sense! Thanks!